pkg/registry/apps/deployment/storage/storage.go

```diff
@@ -376,3 +347,62 @@ func scaleFromDeployment(deployment *apps.Deployment) (*autoscaling.Scale, error
 		},
 	}, nil
 }
+
+// scaleUpdatedObjectInfo transforms existing deployment -> existing scale -> new scale -> new deployment
+type scaleUpdatedObjectInfo struct {
+	name       string
+	reqObjInfo rest.UpdatedObjectInfo
+}
+
+func (i *scaleUpdatedObjectInfo) Preconditions() *metav1.Preconditions {
+	return i.reqObjInfo.Preconditions()
+}
+
+func (i *scaleUpdatedObjectInfo) UpdatedObject(ctx context.Context, oldObj runtime.Object) (runtime.Object, error) {
+	deployment, ok := oldObj.DeepCopyObject().(*apps.Deployment)
+	if !ok {
+		return nil, errors.NewBadRequest(fmt.Sprintf("expected existing object type to be Deployment, got %T", deployment))
+	}
+	// if zero-value, the existing object does not exist
+	if len(deployment.ResourceVersion) == 0 {
+		return nil, errors.NewNotFound(apps.Resource("deployments/scale"), i.name)
+	}
+
+	// deployment -> old scale
+	oldScale, err := scaleFromDeployment(deployment)
+	if err != nil {
+		return nil, err
+	}
+
+	// old scale -> new scale
+	newScaleObj, err := i.reqObjInfo.UpdatedObject(ctx, oldScale)
+	if err != nil {
+		return nil, err
+	}
+	if newScaleObj == nil {
+		return nil, errors.NewBadRequest(fmt.Sprintf("nil update passed to Scale"))
+	}
+	scale, ok := newScaleObj.(*autoscaling.Scale)
+	if !ok {
+		return nil, errors.NewBadRequest(fmt.Sprintf("expected input object type to be Scale, but %T", newScaleObj))
+	}
+
+	// validate
+	if errs := autoscalingvalidation.ValidateScale(scale); len(errs) > 0 {
+		return nil, errors.NewInvalid(autoscaling.Kind("Scale"), deployment.Name, errs)
+	}
+
+	// validate precondition if specified (resourceVersion matching is handled by storage)
+	if len(scale.UID) > 0 && scale.UID != deployment.UID {
+		return nil, errors.NewConflict(
+			apps.Resource("deployments/scale"),
+			deployment.Name,
+			fmt.Errorf("Precondition failed: UID in precondition: %v, UID in object meta: %v", scale.UID, deployment.UID),
```

Comment:
should the error still begin with lower case here?
(applies to the other object types's storage / storage_test)

Reply:
I matched precondition error message format used elsewhere for consistency